### PR TITLE
Add authentication scheme verification

### PR DESCRIPTION
**Pull request by matthieubosquet**
The authentication scheme should be checked as per @RubenVerborgh's comment: https://github.com/solid/access-token-verifier/issues/17#issuecomment-985393932

I kept this PR minimal by just adding the authentication scheme verification.

I will refactor extracting the token parts in a following PR.

### CODE REVIEW

File: src/algorithm/verifyAuthenticationScheme.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * > credentials = "Bearer" 1*SP b64token
+ * > -- https://datatracker.ietf.org/doc/html/rfc6750#section-2.1
+ *
+ * > credentials = "DPoP" 1*SP token68
+ * > -- https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-04#section-7.1
+ *
+ * @param authorizationHeader The authorization header used for the request.
+ */
+export function verifyAuthenticationScheme(authorizationHeader: string): void {
+  if (!/^(DPoP|Bearer) +/i.test(authorizationHeader)) {
```
Comment (RubenVerborgh) [inline src/algorithm/verifyAuthenticationScheme.ts:20]: Technically, we don't need the `+`

File: test/algorithm/verifyAuthenticationScheme.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    }).not.toThrow();
+  });
+
+  it("throws when the authentication scheme is not supported", () => {
+    expect(() => {
+      verifyAuthenticationScheme("dpop");
+      verifyAuthenticationScheme("BeArer");
+      verifyAuthenticationScheme("DPoP\t    ");
+      verifyAuthenticationScheme("bearere             ");
+    }).toThrow();
```
Comment (RubenVerborgh) [inline test/algorithm/verifyAuthenticationScheme.test.ts:19]: Throw what?

**Review by RubenVerborgh — APPROVED** [review]
I find the introduction of `verifyAuthenticationScheme` a bit heavy handed, because the exact same can be checked on line 36 of `verifySolidAccessToken` with one single move. But the functionality is correct.

File: test/algorithm/verifyAuthenticationScheme.test.ts (hunk 2 above)
Comment (matthieubosquet) [reply test/algorithm/verifyAuthenticationScheme.test.ts]: An `AuthenticationSchemeVerificationError` of course!

https://github.com/solid/access-token-verifier/pull/19/commits/bbddfb4c90499468ee4d68d2d7c4c29d97576f72

File: src/algorithm/verifyAuthenticationScheme.ts (hunk 1 above)
Comment (matthieubosquet) [reply src/algorithm/verifyAuthenticationScheme.ts]: That's right.

https://github.com/solid/access-token-verifier/pull/19/commits/03d1e6d518c7287e13dca3624ece69066cad8ea9

**Comment by matthieubosquet** [comment]
> I find the introduction of verifyAuthenticationScheme a bit heavy handed, because the exact same can be checked on line 36 of verifySolidAccessToken with one single move. But the functionality is correct.

My plan is to extract this + token splitting in one function in the next PR.

**Pull request merged**